pkg/services/rendering/auth.go

```diff
@@ -18,15 +24,52 @@ type RenderUser struct {
 	OrgRole string
 }
 
+type renderJWT struct {
+	RenderUser *RenderUser
+	jwt.RegisteredClaims
+}
+
 func (rs *RenderingService) GetRenderUser(ctx context.Context, key string) (*RenderUser, bool) {
-	val, err := rs.RemoteCacheService.Get(ctx, fmt.Sprintf(renderKeyPrefix, key))
+	var (
+		from   string
+		exists bool
+		err    error
+		start  = time.Now()
+	)
+
+	defer func() {
+		success := strconv.FormatBool(err == nil && exists)
+		metrics.MRenderingUserLookupSummary.WithLabelValues(success, from).Observe(float64(time.Since(start)))
+	}()
+
+	if looksLikeJWT(key) && rs.features.IsEnabled(featuremgmt.FlagRenderingOverJWT) {
```

Comment:
Ideally I'd prefer to completely replace any remote cache based auth by JWT-based (when the FF is enabled), but I'm not sure how much could it impact the dashboard thumbnails crawler if we create a new JWT on every request _(also thought about somehow keeping the token in-memory and re-using it until it's close to expire, but sounds like bit of overengineering and a potential can of worms)_.

Any thought from your side @ArturWierzbicki (since looks like you added that feature)?
cc/ @spinillos @undef1nd 

Alternatively, we can just try and use the metric I've just added to evaluate it & decide.

Reply:
creating and signing a jwt is pretty lightweight, I'd start with that and then potentially add an in-memory cache in the crawler that can either get a new key or return the cached one if it's still valid.  +1 on getting rid of the remote cache

Reply:
Do you mean, leaving it as-is, or use JWTs for the crawler as well from now?

Thanks for the suggestions!

Reply:
I mean using JWT for everything (in a followup PR)